### ISSUE TITLE
An adequate Replacement for the Central Mediawiki File Management is needed

### ISSUE THREAD

**Issue by sslHello** [OWASP/owasp.github.io#21]
In my opinion we do need a good replacement for the mediawiki file management.
There is a need to provide OWASP-wide File management to avoid locally managed duplicates and a mass of unknown versions from the same document.
Links to internal OWASP-documents should look dirrerent from external links. 
Please deploy a solution and give us guidelines how to use it.

I am sorry to say, but in my opinion this is a critical factor for any migration.

**Comment by hblankenship** [comment]
Hey Torsten.  What sort of documents are you specifically referring to?  Are you meaning simply a place to store non-web-specific (not md, html) files from which others could reference the files?  Some of this exists already with regard to images which can be used in projects, etc.

Also, the ability to add files to the www-community repository is open to all.  There is a folder called assets in that repository and folders under that for files and images.

**Comment by sslHello** [comment]
Yes the issue is mainly about non-web-specific files, e.g. PDFs, pictures etc that are commonly used/linked, e.g.:
https://www.owasp.org/index.php/File:OWASP_Project_Header.jpg
https://www.owasp.org/index.php/File:Flagship_big.jpg
Or a PDF used by different project and chapter pages:
https://www.owasp.org/index.php?title=File:OWASP_Top_10-2017_de_V1.0.pdf

Additionally some pages transclude even other web-files:
https://www.owasp.org/index.php/GPC_Project_Details/OWASP_Top10
see usage: https://www.owasp.org/index.php/Special:WhatLinksHere/GPC_Project_Details/OWASP_Top10
So, you can use the same content to be used from different pages.

This gives us the opportunity to define reusable design, pages and pictures that can be centrally managed. There is an huge number of pages using this.
Cheers Torsten

**Comment by sslHello** [comment]
Hi Harold,
I am sorry, may I ask if there is any progess on this?
Thank you so much.
Cheers Torsten

**Comment by hblankenship** [comment]
Hey Torsten,

With regard to uploading files that are meant to be used/linked across
repositories, you can either create a pull request to include them in
github.owasp.io (if that would make sense) or, more likely, under
www-community repository (under assets/files)


Harold L. Blankenship

OWASP Foundation

Director of Technology and Projects

**Comment by kingthorin** [comment]
Image type assets to be used globally across owasp.org should be added to `OWASP/www--site-theme` (that's where the existing status and builder/breaker/defender images are.

For PDFs (or other 'download' type files... @hblankenship correct me if I'm wrong) there's `OWASP/www-pdf-archive`.

**Comment by sslHello** [comment]
Yes, for central storage, I had more files in mind that are communality used, eg. Icons.
Surplus the wiki offered a file management, e.g. to see 'where used' to see the file has been used. The same for templates....  
I haven't seen any replacement for this, yet,.

**Comment by akshit5565** [comment]
@vanderaj can you please assign this task to me?

**Comment by kingthorin** [comment]
What are you planning as a solution?
